src/components/dac_dataset_table/DACDatasetTable.js

```diff
@@ -0,0 +1,173 @@
+};
+
+const columnHeaderConfig = {
+  duosId: {label: 'DUOS ID', cellStyle: {width: styles.cellWidths.duosId}, cellDataFn: cellData.duosIdCellData, sortable: true},
+  dataSubmitter: {label: 'Data Submitter', cellStyle: {width: styles.cellWidths.dataSubmitter}, cellDataFn: cellData.dataSubmitterCellData, sortable: true},
+  datasetName: {label: 'Dataset Name', cellStyle: {width: styles.cellWidths.datasetName}, cellDataFn: cellData.datasetNameCellData, sortable: true},
+  dataCustodian: {label: 'Data Custodian', cellStyle: {width: styles.cellWidths.dataCustodian}, cellDataFn: cellData.dataCustodianCellData, sortable: true},
+  dataUse: {label: 'Data Use', cellStyle: {width: styles.cellWidths.dataUse}, cellDataFn: cellData.dataUseCellData, sortable: false},
+  status: {label: 'Status', cellStyle: {width: styles.cellWidths.status}, cellDataFn: cellData.statusCellData, sortable: false},
+};
+
+const defaultColumns = Object.keys(columnHeaderConfig);
+
+const columnHeaderData = (columns = defaultColumns) => {
+  return columns.map((col) => columnHeaderConfig[col]);
+};
+
+const processDatasetRowData = ({
+  datasets, columns = defaultColumns, consoleType=''
+}) => {
+  if(!isNil(datasets)) {
+    return datasets.map((dataset) => {
+      return columns.map((col) => {
+        return columnHeaderConfig[col].cellDataFn({
+          dataset, consoleType
+        });
+      });
+    });
+  }
+};
+
+const storageDACDatasetSort = 'storageDACDatasetSort';
+
+const getInitialSort = (columns = []) => {
+  const sort = Storage.getCurrentUserSettings(storageDACDatasetSort) || {
+    field: DACDatasetTableColumnOptions.DUOS_ID,
+    dir: -1
+  };
+  const sortIndex = columns.indexOf(sort.field);
+
+  if (sortIndex !== -1) {
+    return { colIndex: sortIndex, dir: sort.dir};
+  }
+  else {
+    return { colIndex: 0, dir: 1 };
+  }
+};
+
+export const DACDatasetTable = function DACDatasetTable(props) {
+  const [visibleDatasets, setVisibleDatasets] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pageCount, setPageCount] = useState(1);
+  const [sort, setSort] = useState(getInitialSort(props.columns));
+  const [tableSize, setTableSize] = useState(10);
+  const { datasets, columns, isLoading, consoleType } = props;
+
+  const changeTableSize = useCallback((value) => {
+    if (value > 0 && !isNaN(parseInt(value))) {
+      setTableSize(value);
+    }
+  }, []);
+
+  useEffect(() => {
+    recalculateVisibleTable({
+      tableSize,
+      pageCount,
+      filteredList: processDatasetRowData({datasets, columns, consoleType}),
+      currentPage,
+      setPageCount,
+      setCurrentPage,
+      setVisibleList: setVisibleDatasets,
+      sort,
+    });
+  }, [tableSize, currentPage, pageCount, datasets, sort, columns, consoleType]);
+
+  //Helper function to update page
+  const goToPage = useCallback(
+    (value) => {
+      updatePage(value, pageCount, setCurrentPage);
+    },
+    [pageCount]
+  );
+
+  return h(SimpleTable, {
```

Comment:
It looks like this table might benefit from some style modifications:
1) When the table is narrow, cell data is spilling out of the "cell" and overlaps contents from another cell.  Ideally, we should have a margin between cells.
2) We might want to consider using ellipses for long cell content.